File: docs/WidgetMill/packages/spec/src/nodes/sizedBox.ts
import { z } from "zod";
import { BindableDimension } from "../complex/dimension";
import { NodeRef } from "../node-ref";

/** SizedBox — spec §2.9. */
export const SizedBoxNode = z.object({
  type: z.literal("sizedBox"),
  props: z
    .object({
      width: BindableDimension.optional(),
      height: BindableDimension.optional(),
    })
    .default({}),
  child: NodeRef.optional(),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/sized_box.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/parsers.dart';
import '../renderer.dart';

/// SizedBox — spec §2.9.
Widget buildSizedBox(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return SizedBox(
    width: resolveDimension(context, p['width']),
    height: resolveDimension(context, p['height']),
    child: r.maybeRender(context, node.child),
  );
}
